loki/frontend/tests/test_frontends.py

```diff
@@ -2020,6 +2020,69 @@ def to_str(expr):
     assert isinstance(routine.variable_map['v6'].shape[1], sym.Power)
     assert isinstance(routine.variable_map['v6'].shape[2], sym.Quotient)
 
+@pytest.mark.parametrize('frontend', available_frontends(xfail=[(OMNI, 'OMNI does not like Loki pragmas, yet!')]))
+def test_frontend_module_variables_dimension_pragmas(frontend, tmp_path):
+    """
+    Test that `!$loki dimension` pragmas can be used to override the
+    conceptual `.shape` of module variables.
+    """
+    code_mod = """
+module mod_variable_dimensions
+
+  integer, parameter :: jprb = selected_real_kind(13,300)
+  integer :: x, y
+
+  !$loki dimension(10)
+  real(kind=jprb), intent(inout) :: v0(:)
+  !$loki dimension(x)
+  real(kind=jprb), intent(inout) :: v1(:)
+  !$loki dimension(x,y,:)
+  real(kind=jprb), dimension(:,:,:), intent(inout) :: v2, v3
+  !$loki dimension(x,y)
+  real(kind=jprb), pointer, intent(inout) :: v4(:,:)
+  !$loki dimension(x+y,2*x)
+  real(kind=jprb), allocatable :: v5(:,:)
+  !$loki dimension(x/2, x**2, (x+y)/x)
+  real(kind=jprb), dimension(:, :, :), pointer :: v6 
+end module mod_variable_dimensions
+    """
+
+    def to_str(expr):
+        return str(expr).lower().replace(' ', '')
```

Comment:
This looks like we're missing a StrCompareMixin somewhere?

Reply:
Yes, but that's an (internally) know issue; we don't force `StrCompareMixin` on pure operations (`Sum`, `Product`, etc.). We could, probably, but beyond this PR I'd argue? Maybe we need a separate (formal) issue on this though?

Reply:
Yes fully agreed - I didn't mean to imply this needs to be actioned here. Documenting this in an issue would be a good idea, though.